### ISSUE TITLE
x/tools/gopls: typeDefinitioon

### ISSUE THREAD

**Issue by adonovan** [golang/go#67890]

**Comment by vikblom** [comment]
Hi, I'd be interested in helping out with this, although I am a bit fuzzy on what should be added.

As far as I know I only use typeDefinition as a goto/jump. 

Playing around with gopls v0.16.0 it seems like struct fields (in definitions, assignment, and access) can be "jumped from". Is there a case I'm missing?

The same goes for an expression like
```
s.view.importsState.runProcessEnvFunc(ctx, s, fn)
```
where the snapshot, view and importState types can all be reached. Although the type of the function return, which is `error`, does _not_ work, so perhaps that is a good place to start?

**Comment by findleyr** [comment]
@vikblom indeed, struct fields *do* work. I think that decreases the utility of this issue significantly; jumping to the type definition of call expressions can be achieved by jumping on the function name, since it traverses through the structure of simple expressions to find the "element" type.

Perhaps https://go.dev/issue/66557, which you've already identified, is a better place for a next contribution.
